src/devices/src/virtio/vsock/unix/muxer.rs

```diff
@@ -776,6 +777,17 @@ mod tests {
         }
     }
 
+    // Create a TempFile with a given prefix and return it as a nice String
+    fn get_file(fprefix: &str) -> String {
+        let listener_path = TempFile::new_with_prefix(fprefix.to_owned()).unwrap();
+        listener_path
+            .as_path()
+            .as_os_str()
+            .to_str()
+            .unwrap()
+            .to_owned()
+    }
```

Comment:
This will indeed use a unique path, but will not cleanup the socket UDS when the test is complete.

The TempFile here will cleanup itself when it goes out of scope (end of this fn), the path is passed to the muxer which creates an UDS there, but nobody cleans it up.

You can check `src/vmm/src/vmm_config/vsock.rs` tests to see an example to make sure it's also cleaned up.

Ideally we should create a `TempPathObject` in some tests utility crate so we can reuse the model everywhere instead of duplicating it around (it's currently duplicated in a few places around our codebase).

```rust
    /// When this object goes out of scope, the internal path is `unlinked` from the file-system.
    pub(crate) struct TempPathObject {
        path: String,
    }
    impl TempPathObject {
        pub fn new(fprefix: &str) -> Self {
            let obj_path = TempFile::new_with_prefix(fprefix.to_owned()).unwrap();
            TempPathObject {
                path: obj_path
                    .as_path()
                    .as_os_str()
                    .to_str()
                    .unwrap()
                    .to_owned(),
            }
        }
        pub fn path(&self) -> &String {
            &self.path
        }
    }
    impl Drop for TempPathObject {
        fn drop(&mut self) {
            let _ = std::fs::remove_file(&self.path);
        }
    }
```

Reply:
> This will indeed use a unique path, but will not cleanup the socket UDS when the test is complete.

Line `774` implements the `Drop` trait for `MuxerTestContext`.

In the ideal scenario where this test works and finishes successfully, the `host_sock_path` is cleaned up and another test run can be executed.
If the test panics for some reason, there will be files left in the test folder.

The reason for dropping the `listener_path` variable there is that the `VsockMuxer` tries to open the file again and since the `TempPath` implementation opens the file by default, we just need a random unopened path.

Reply:
>  If the test panics for some reason, there will be files left in the test folder.

Drop is called on panic as well, so we shouldn't have this problem.

Reply:
You're right @gc-plp, the UDS is in fact cleaned up by the MuxerTestContext drop impl.
I'm resolving this comment, feel free to merge the PR.